fe/fe-core/src/main/java/com/starrocks/meta/lock/LightWeightLock.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.meta.lock;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class LightWeightLock extends Lock {
+    private LockHolder lockHolder;
+
+    public LightWeightLock() {
+    }
+
+    @Override
+    public LockGrantType lock(Locker locker, LockType requestLockType) {
+        assert lockHolder == null;
+        this.lockHolder = new LockHolder(locker, requestLockType);
+        return LockGrantType.NEW;
+    }
+
+    @Override
+    public Set<Locker> release(Locker locker, LockType lockType) throws NotSupportLockException {
+        if (lockHolder.equals(new LockHolder(locker, lockType))) {
+            this.lockHolder = null;
+            return null;
+        } else {
+            throw new NotSupportLockException("Attempt to unlock lock, not locked by current locker");
```

Comment:
Throws:
IllegalMonitorStateException – if the current thread does not hold this lock